sda-commons-server-dropwizard/src/main/java/org/sdase/commons/server/dropwizard/bundles/SystemPropertyAndEnvironmentLookup.java

```diff
@@ -1,19 +1,78 @@
 package org.sdase.commons.server.dropwizard.bundles;
 
-import io.dropwizard.configuration.EnvironmentVariableLookup;
+import com.fasterxml.jackson.core.io.CharTypes;
+import java.util.Optional;
+import org.apache.commons.text.lookup.StringLookup;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * Lookup for Java's system properties and environment variables. System properties have higher
  * priority.
  */
-public class SystemPropertyAndEnvironmentLookup extends EnvironmentVariableLookup {
+public class SystemPropertyAndEnvironmentLookup implements StringLookup {
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(SystemPropertyAndEnvironmentLookup.class);
+
+  private static final String TO_JSON_STRING_OPERATOR_NORMALIZED = "toJsonString".toLowerCase();
+
+  /**
+   * Provides the actual key to lookup without any operators.
+   *
+   * @param definedKey The key with operators as defined in the String with substitutions. Examples:
+   *     <ul>
+   *       <li><code>foo: ${FOO_PROPERTY}</code> =&gt; definedKey = {@code FOO_PROPERTY}
+   *       <li><code>foo: ${FOO_PROPERTY:-default}</code> =&gt; definedKey = {@code FOO_PROPERTY}
+   *       <li><code>foo: ${FOO_PROPERTY | toJsonString:-default}</code> =&gt; definedKey = {@code
+   *           FOO_PROPERTY | toJsonString}
+   *     </ul>
+   *
+   * @return The key that should be looked up in the environment. All examples for {@code
+   *     definedKey} above will return {@code FOO_PROPERTY}.
```

Comment:
Your documentation says: "All examples will return `FOO_PROPERTY`". But if I understand your last example right, you also return the operator. Which documentation is correct?

Reply:
It picks only the first item (the actual key) of the split.